rotkehlchen/tests/exchanges/test_bitfinex.py

```diff
@@ -0,0 +1,1376 @@
+
+    get_response = get_paginated_response()
+    currency_map_patch = patch.object(
+        target=mock_bitfinex,
+        attribute='_query_currency_map',
+        return_value=currency_map_response,
+    )
+    api_limit_patch = patch(
+        target='rotkehlchen.exchanges.bitfinex.API_MOVEMENTS_MAX_LIMIT',
+        new=api_limit,
+    )
+    api_query_patch = patch.object(
+        target=mock_bitfinex,
+        attribute='_api_query',
+        side_effect=mock_api_query_response,
+    )
+    with ExitStack() as stack:
+        stack.enter_context(api_limit_patch)
+        stack.enter_context(currency_map_patch)
+        stack.enter_context(api_query_patch)
+        asset_movements = mock_bitfinex.query_online_deposits_withdrawals(
+            start_ts=Timestamp(0),
+            end_ts=Timestamp(int(datetime.now().timestamp())),
+        )
+        wbtc_fee_asset = Asset('WBTC')
+        eur_fee_asset = Asset('EUR')
+        expected_asset_movements = [
+            AssetMovement(
+                timestamp=Timestamp(1606899600),
+                location=Location.BITFINEX,
+                category=AssetMovementCategory.DEPOSIT,
+                address='DESTINATION_ADDRESS',
+                transaction_id='TRANSACTION_ID',
+                asset=wbtc_fee_asset,
+                amount=FVal('0.26300954'),
+                fee_asset=wbtc_fee_asset,
+                fee=Fee(FVal('0.00135')),
+                link=str(1),
+            ),
+            AssetMovement(
+                timestamp=Timestamp(1606901400),
+                location=Location.BITFINEX,
+                category=AssetMovementCategory.WITHDRAWAL,
+                address='DESTINATION_ADDRESS',
+                transaction_id='TRANSACTION_ID',
+                asset=wbtc_fee_asset,
+                amount=FVal('0.26300954'),
+                fee_asset=wbtc_fee_asset,
+                fee=Fee(FVal('0.00135')),
+                link=str(2),
+            ),
+            AssetMovement(
+                timestamp=Timestamp(1606986000),
+                location=Location.BITFINEX,
+                category=AssetMovementCategory.WITHDRAWAL,
+                address=None,
+                transaction_id=None,
+                asset=eur_fee_asset,
+                amount=FVal('0.26300954'),
+                fee_asset=eur_fee_asset,
+                fee=Fee(FVal('0.00135')),
+                link=str(3),
+            ),
+            AssetMovement(
+                timestamp=Timestamp(1606996800),
+                location=Location.BITFINEX,
+                category=AssetMovementCategory.DEPOSIT,
+                address='DESTINATION_ADDRESS',
+                transaction_id='TRANSACTION_ID',
+                asset=wbtc_fee_asset,
+                amount=FVal('0.26300954'),
+                fee_asset=wbtc_fee_asset,
+                fee=Fee(FVal('0.00135')),
+                link=str(4),
+            ),
+        ]
+        assert asset_movements == expected_asset_movements
```

Comment:
Wow that's A LOT of tests. Cant' comment on their validity, but where did you get the test data from? API docs?  Or your own data? It's hard to be sure stuff works if you are just testing their docs.

Reply:
Of course from their docs... I have no data :(

Are not a lot imho:
- API error codes/messages (some don't have codes) work as expected. Especially manipulating the response as a list.
- Balances/Trades/AssetMovement
- Paginations
- Discover new exchange assets

Reply:
Perhaps it's just the amount of lines due to the way we split the arguments. Anyway no worries. With whatever you learn from the debugging session with Phil, please adjust tests or add new ones.